R/mixture_model_wrapper.R

```diff
@@ -0,0 +1,141 @@
+#' Stack models according to CRPS
+#'
+#' @description Provides a wrapper for different EpiSoon models and generates
+#' a mixture model based on these models
+#'
+#' @param models A list of models. Models must be analogous to the form
+#' `function(...){EpiSoon::fable_model(model = , ...)}` or
+#' `function(...){EpiSoon::bsts_model(model = , ...)}`.
+#' @inheritParams bsts_model
+#' @return A dataframe of predictions (with columns representing the
+#'  time horizon and rows representing samples).
+#' @export
+#' @importFrom tsibble tsibble
+#' @importFrom fabletools model forecast
+#' @importFrom purrr map
+#' @importFrom dplyr bind_rows
+#' @examples
+#'
+#' y <- EpiSoon::example_obs_rts[1:10, ]$rt
+#' model = fable::ARIMA(y ~ time)
+#' samples = 10
+#' horizon = 7
+#' weighting_period = 5
+#'
+#' # make list with models
+#' baseline_ensemble <- function(...) {
+#' EpiSoon::fable_model(model = fabletools::combination_model(fable::ARIMA(y), fable::ETS(y), fable::NAIVE(y),
+#'                                                            fable::RW(y ~ drift()), cmbn_args = list(weights = "inv_var")), ...)
+#' }
+#' models <- list(
+#'   Mixfabl = baseline_ensemble,
+#'   "ARIMA" = function(...){EpiSoon::fable_model(model = fable::ARIMA(y), ...)},
+#'   "ETS" = function(...){EpiSoon::fable_model(model = fable::ETS(y), ...)},
+#'   "Drift" = function(...){EpiSoon::fable_model(model = fable::RW(y ~ drift()), ...)}
+#' )
+#'
+#' # make forecst
+#' forecast <- crps_ensemble(y = y,
+#'                           models = models,
+#'                           samples = 10,
+#'                           horizon = 7,
+#'                           weighting_period = 5)
+#'
+#'
+#'
+#' forecast_rt(EpiSoon::example_obs_rts[1:10, ],
+#'            model = function(...){
+#'              crps_ensemble(y = y,
```

Comment:
I think to work comparably to other models this needs to accept the samples, y and horizon not have them embed at this stage. Think this will fail when nested in evaluate model etc.